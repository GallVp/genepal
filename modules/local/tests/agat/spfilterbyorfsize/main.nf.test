nextflow_process {

    name "Test Process AGAT_SPFILTERBYORFSIZE"
    script "../../../../nf-core/agat/spfilterbyorfsize/main.nf"
    process "AGAT_SPFILTERBYORFSIZE"

    tag "modules"
    tag "modules_nfcore"
    tag "agat"
    tag "agat/spfilterbyorfsize"

    test("Knockout one transcript") {

        when {
            process {
                """
                input[0] = [
                    [id: 'test'],
                    file("$baseDir" + '/modules/local/tests/gffread/testdata/t.gff', checkIfExists: true)
                ]
                input[1] = []
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
            with (file(process.out.passed_gff[0][1]).text) {
                assert contains('gene19851')
                assert contains('gene19851.t1')

                // This is the only transcript which should be knocked out
                assert ! contains('gene19851.t2')
            }
        }

    }

    test("Knockout long transcript with short CDS") {

        when {
            process {
                """
                input[0] = [
                    [id: 'test'],
                    file("$baseDir" + '/modules/local/tests/gffread/testdata/t2.gff', checkIfExists: true)
                ]
                input[1] = []
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
            with (file(process.out.passed_gff[0][1]).text) {
                assert contains('P005ME_h2_v2.g24')

                // The following transcripts should be knocked out because AGAT considers the CDS length
                // as opposed to the length of the transcript
                assert ! contains('P005ME_h2_v2.g14880')
                assert ! contains('P005ME_h2_v2.g19822')
                assert ! contains('P005ME_h2_v2.g8694')
            }
        }

    }

}
