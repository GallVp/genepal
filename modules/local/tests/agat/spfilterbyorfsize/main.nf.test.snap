{
    "Knockout long transcript with short CDS": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.passed.gff:md5,a33093410a39dde09385b5ae599b0bc9"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.failed.gff:md5,07786c2cae48fa668828c41dae118d45"
                    ]
                ],
                "2": [
                    "versions.yml:md5,a3d1e735adde2ea534a2ca079be3c543"
                ],
                "failed_gff": [
                    [
                        {
                            "id": "test"
                        },
                        "test.failed.gff:md5,07786c2cae48fa668828c41dae118d45"
                    ]
                ],
                "passed_gff": [
                    [
                        {
                            "id": "test"
                        },
                        "test.passed.gff:md5,a33093410a39dde09385b5ae599b0bc9"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,a3d1e735adde2ea534a2ca079be3c543"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-07T11:01:04.86992"
    },
    "Knockout one transcript": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.passed.gff:md5,05ccb11ad8fcec5cb070ae0e8d5b4091"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.failed.gff:md5,131fa308127fa6006a86be3b86c574df"
                    ]
                ],
                "2": [
                    "versions.yml:md5,a3d1e735adde2ea534a2ca079be3c543"
                ],
                "failed_gff": [
                    [
                        {
                            "id": "test"
                        },
                        "test.failed.gff:md5,131fa308127fa6006a86be3b86c574df"
                    ]
                ],
                "passed_gff": [
                    [
                        {
                            "id": "test"
                        },
                        "test.passed.gff:md5,05ccb11ad8fcec5cb070ae0e8d5b4091"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,a3d1e735adde2ea534a2ca079be3c543"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-07T11:01:00.710911"
    }
}